sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzMonListExtensions.cs

```diff
@@ -291,12 +305,13 @@ internal static string GetDefaultDbPort(string? dbSystem)
         ///</summary>
         internal static (string? DbName, string? DbTarget) GetDbDependencyTargetAndName(this AzMonList tagObjects)
         {
-            string? target = null;
-            string defaultPort = GetDefaultDbPort(AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeDbSystem)?.ToString());
-            var peerService = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributePeerService)?.ToString();
-            if (!string.IsNullOrWhiteSpace(peerService))
+            var peerServiceAndDbSystem = AzMonList.GetTagValues(ref tagObjects, SemanticConventions.AttributePeerService, SemanticConventions.AttributeDbSystem);
+            string? target = peerServiceAndDbSystem[0]?.ToString();
+            var defaultPort = GetDefaultDbPort(peerServiceAndDbSystem[1]?.ToString());
+
+            if (string.IsNullOrWhiteSpace(target))
```

Comment:
nit: This might improve readability of the code.

```
    if (string.IsNullOrWhiteSpace(target))
    {
        target = tagObjects.GetTargetUsingServerAttributes(defaultPort) ?? tagObjects.GetTargetUsingNetPeerAttributes(defaultPort);
    }
```